src/trackers/helpers/nameserver.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Handle name server lookups
+ */
+const dns = require('dns').promises
+const fs = require('fs')
+const cache = {}
+
+class NameServers {
+    /**
+     * Attempt to look name servers for a given hostname
+     * @param {string} host - hostname to look up.
+     *
+     * @return {array} The name server resolution
+     */
+    static async resolveNs(host) {
+        if (!host) {
+            return undefined
+        }
+
+        if (host in cache) {
+            return cache[host]
+        }
+       
+        // New DNS NS request
+        // Sites are processed in batches. It's possible for multiple NS requests to 
+        // happen before the cache has a resolved NS response. Handle this by adding the unresolved 
+        // promise to the cache so multiple requests for the same host can reference the 
+        // same cache entry
+        cache[host] = dns.resolveNs(host).catch(e => _handleNsError(e, host))
+        cache[host] = await cache[host]
```

Comment:
You don't need this line. As this is an async function it will always something Promise-like, so you can just return the original promise instead of awaiting it.

Reply:
👍 thanks